modules/apps/headless/headless-delivery/headless-delivery-test/src/testIntegration/java/com/liferay/headless/delivery/resource/v1_0/test/MessageBoardMessageResourceTest.java

```diff
@@ -46,11 +48,49 @@ public void setUp() throws Exception {
 		serviceContext.setScopeGroupId(testGroup.getGroupId());
 	}
 
+	@Override
+	@Test
+	public void testPutMessageBoardThreadMessageBoardMessage()
+		throws Exception {
+
+		// Update
+
+		super.testPutMessageBoardThreadMessageBoardMessage();
+
+		// Insert
+
+		MessageBoardMessage randomMessageBoardMessage =
+			randomMessageBoardMessage();
+
+		randomMessageBoardMessage.setMessageBoardThreadId(
+			_mbThread.getThreadId());
+
+		MessageBoardMessage insertMessageBoardMessage =
```

Comment:
insert is strange when you are doing a PUT to do an upsert, is it used anywhere? why not upsert/new/_ directly?

Reply:
I was differenciating between the two possible cases of the upsert. The update when the externalReferenceCode exists that is covered by the base method. And the insert when the externalReferenceCode does not exist.

But you are right that maybe new would have been a better name.

It is only used here on the tests

Reply:
yeah, that's why I gave a 👍